mmv1/third_party/terraform/services/resourcemanager/ephemeral_google_service_account_id_token.go

```diff
@@ -0,0 +1,173 @@
+				Optional:    true,
+				ElementType: types.StringType,
+				Validators: []validator.Set{
+					setvalidator.ValueStringsAre(fwvalidators.ServiceAccountEmailValidator{}),
+				},
+			},
+			"include_email": schema.BoolAttribute{
+				Description: "Include the verified email in the claim. Used only when using impersonation mode.",
+				Optional:    true,
+				Computed:    true,
+			},
+			"id_token": schema.StringAttribute{
+				Description: "The `id_token` representing the new generated identity.",
+				Computed:    true,
+				Sensitive:   true,
+			},
+		},
+	}
+}
+
+func (p *googleEphemeralServiceAccountIdToken) Configure(ctx context.Context, req ephemeral.ConfigureRequest, resp *ephemeral.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	pd, ok := req.ProviderData.(*fwtransport.FrameworkProviderConfig)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Data Source Configure Type",
+			fmt.Sprintf("Expected *fwtransport.FrameworkProviderConfig, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+		return
+	}
+
+	// Required for accessing userAgent and passing as an argument into a util function
+	p.providerConfig = pd
+}
+
+func (p *googleEphemeralServiceAccountIdToken) Open(ctx context.Context, req ephemeral.OpenRequest, resp *ephemeral.OpenResponse) {
+	var data ephemeralServiceAccountIdTokenModel
+
+	resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
+
+	targetAudience := data.TargetAudience.ValueString()
+
+	// TODO: This is a temporary solution to get the credentials from the provider config.
+	// we'll address this once muxing issues are resolved.
+	model := fwmodels.ProviderModel{
+		Credentials:                        p.providerConfig.Credentials,
+		AccessToken:                        p.providerConfig.AccessToken,
+		ImpersonateServiceAccount:          p.providerConfig.ImpersonateServiceAccount,
+		ImpersonateServiceAccountDelegates: p.providerConfig.ImpersonateServiceAccountDelegates,
+		Project:                            p.providerConfig.Project,
+		BillingProject:                     p.providerConfig.BillingProject,
+		Scopes:                             p.providerConfig.Scopes,
+		UniverseDomain:                     p.providerConfig.UniverseDomain,
+	}
+	creds := fwtransport.GetCredentials(ctx, model, false, &resp.Diagnostics)
+	targetServiceAccount := data.TargetServiceAccount
+	// If a target service account is provided, use the API to generate the idToken
+	if !targetServiceAccount.IsNull() && !targetServiceAccount.IsUnknown() {
+		service := p.providerConfig.NewIamCredentialsClient(p.providerConfig.UserAgent)
+		name := fmt.Sprintf("projects/-/serviceAccounts/%s", targetServiceAccount.ValueString())
+
+		tokenRequest := &iamcredentials.GenerateIdTokenRequest{
+			Audience:     targetAudience,
+			IncludeEmail: data.IncludeEmail.ValueBool(),
```

Comment:
Re: my last review, I think if include email was left unset (ie was Null) then this would result in a false being used here. Could you please confirm that?